frontend/tests/hooks/useSimulationQueriesResult.test.tsx

```diff
@@ -0,0 +1,139 @@
+﻿import { describe, it, expect, vi, beforeEach } from "vitest";
+import { renderHook, waitFor } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ExperimentResult } from "../../src/dto/ExperimentResult";
+import { ModelConfig } from "../../src/dto/FormConfig";
+import { ChartDataSet } from "../../src/dto/ChartData";
+import React from "react";
+
+// Mock data
+const mockExperiments: ExperimentResult[] = [
+    {
+        name: "Experiment 1",
+        time: 100,
+        temperature: 25,
+        pressure: 1,
+        initialConcentrations: { CO2: 0.5, H2O: 0.3 },
+        finalConcentrations: { CO2: 0.4, H2O: 0.4 },
+    },
+    {
+        name: "Experiment 2",
+        time: 200,
+        temperature: 30,
+        pressure: 1.2,
+        initialConcentrations: { CO2: 0.6, H2O: 0.2 },
+        finalConcentrations: { CO2: 0.3, H2O: 0.5 },
+    },
+];
+
+const mockModels: ModelConfig[] = [
+    {
+        modelId: "model1",
+        displayName: "Primary Model 1",
+        validSubstances: ["CO2", "H2O"],
+        parameters: { Temperature: { default: 300 }, Pressure: { default: 1 } },
+        description: "Test model",
+        category: "Primary",
+    },
+    {
+        modelId: "model2",
+        displayName: "Primary Model 2",
+        validSubstances: ["CO2", "H2O", "CH4"],
+        parameters: {},
+        description: "Test model",
+        category: "Primary",
+    },
+];
+
+const mockSimulationResult = {
+    initialConcentrations: { CO2: 0.5, H2O: 0.3 },
+    finalConcentrations: { CO2: 0.45, H2O: 0.35 },
+    panels: [],
+};
+
+const mockChartDataSet: ChartDataSet = {
+    label: "Primary Model 1 (Experiment 1)",
+    data: [
+        { x: "CO2", y: 0.45 },
+        { x: "H2O", y: 0.35 },
+    ],
+};
+
+// Mock external dependencies using factory functions to avoid hoisting issues
+vi.mock("../../src/api/api", () => ({
+    runSimulation: vi.fn(),
+}));
+
+vi.mock("../../src/functions/Filtering", () => ({
+    filterValidModels: vi.fn(),
+}));
+
+vi.mock("../../src/functions/Formatting", () => ({
+    convertSimulationToChartData: vi.fn(),
+}));
+
+vi.mock("../../src/contexts/ModelContext", () => ({
+    useAvailableModels: vi.fn(),
+}));
+
+import { useSimulationQueries } from "../../src/hooks/UseSimulationQueriesResult";
```

Comment:
considering what this does, is `useSimulationQueries` a fitting name? What would we like this hook to do?

We have some data (which happens to be experimental), and we want the hook to run a simulation for all models that can accept the input data and return the results for all models as soon as they are available.

We did discuss caching results, at least we are currently doing it. Is this custom hook the place to do so? (I don't know to be honest). As that would be an additional feature to what it does (and maybe should be reflected in the name).

Reply:
we might also want to consider if this hook can use the context provider of `SimulationProvider`. We will likely need to initiate the `SimulationProvider` at a lower level though (currently it is top level). Should probably be started per page. Then I think we can reuse quite a bit to be honest.